bin/styled-components-codemods.js

```diff
@@ -1,44 +1,75 @@
 #!/usr/bin/env node
-'use strict';
-
-const path = require('path');
-const program = require('commander');
-const { spawn } = require('child_process');
-const pkg = require('../package.json');
-
-program
-  .version(pkg.version)
-  .command('v4 [files...]')
-  .action((files) => {
-    if (!Array.isArray(files) || files.length === 0) {
-      throw new Error('Please provide a list of files to run the codemod on.');
-    } 
-
-    console.log('Running v4 codemods...\n')
-    const child = spawn(path.resolve(path.join(__dirname, '../node_modules/.bin/codemod')), ['--plugin', path.resolve(path.join(__dirname, '../src/v4/index.js')), ...files])
-
-    child.stdout.on('data', chunk => {
-      console.log(chunk.toString())
-    })
-    // since these are streams, you can pipe them elsewhere
-    child.stderr.on('data', chunk => {
-      console.error(chunk.toString());
-    });
+"use strict";
+
+const path = require("path");
+const program = require("commander");
+const { spawnSync } = require("child_process");
+const { curry } = require("lodash");
+const pkg = require("../package.json");
+
+/**
+ * Add the version-specific codemods here.
+ */
+const codemods = {
+  v4: ["extendToStyled", "injectGlobalToCreateGlobalStyle"]
```

Comment:
The upshot of this is all you have to do to wire up another one to commander is add its filename here.

Reply:
I'm thinking later we can standardize on this more (e.g., examples are still disconnected from this structure).